pombola/south_africa/views/attendance.py

```diff
@@ -53,20 +54,30 @@ def filter_attendance(self, annual_attendance, party, position):
             attendance_summary = [ma for ma in attendance_summary if
                 ma['member']['party_name'] == party]
 
-        if position == 'ministers':
-            active_ministers = Position.objects.filter(
+        year_end_date = datetime.datetime.strptime(
+            annual_attendance['end_date'], "%Y-%m-%d").date()
+        active_ministers = Position.objects.filter(
                 Q(title__slug__startswith='minister') |
                 Q(title__slug__startswith='deputy-minister')
-            ).overlapping_dates(
-                annual_attendance['start_date'],
-                annual_attendance['end_date'])
+            ).currently_active(
+                year_end_date)
```

Comment:
Great, thank you for pointing that out. There's going to be a couple of tweaks in the next few days, and I'll simplify it with those changes.